src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -776,22 +768,58 @@ protected Pool<Jedis> createRedisSentinelPool(RedisSentinelConfiguration config)
 	 * @since 1.4
 	 */
 	protected Pool<Jedis> createRedisPool() {
-		return new JedisPool(getPoolConfig(), new HostAndPort(getHostName(), getPort()), this.clientConfig);
+		return new JedisPool(getPoolConfig(), newHostAndPort(getHostName(), getPort()), this.clientConfig);
 	}
 
 	/**
-	 * Template method to create a {@link ClusterTopologyProvider} given {@link JedisCluster}. Creates
-	 * {@link JedisClusterTopologyProvider} by default.
-	 *
-	 * @param cluster the {@link JedisCluster}, must not be {@literal null}.
-	 * @return the {@link ClusterTopologyProvider}.
-	 * @see JedisClusterTopologyProvider
-	 * @see 2.2
+	 * @deprecated Use {@link #createClusterTopologyProvider(JedisCluster)} instead.
+	 * @since 2.2
 	 */
+	@Deprecated(since = "3.2")
```

Comment:
I do not understand what warrants a deprecation of the `createTopologyProvider`. New deprecations are a signal for existing users to do something about their usage while we do not solve an actual problem.

Reply:
Naming consistency with other cluster-oriented types.

Reply:
I was quite surprised when I first learned about Redis that we don't have something equivalent to `SentinelTopologyProvider` for Redis Sentinel.

Of course, Redis Sentinel is not a fully managed, "clustered" solution (only HA), but is a group of nodes in a distributed system none-the-less.

Technically, `ClusterTopology` is `Set` of `RedisClusterNode` and similarly, `RedisSentinelConfiguration` contains a `Set` of `RedisNode` where `RedisClusterNode` extends from `RedisNode`. So, although they are not quite the same, they are similar enough.